src/lib/rules/strict-transport-security/strict-transport-security.ts

```diff
@@ -0,0 +1,128 @@
+// ------------------------------------------------------------------------------
+
+const rule: IRuleBuilder = {
+    create(context: RuleContext): IRule {
+        /** The minimum period (in seconds) allowed for `max-age`. */
+        let minMaxAgeValue: number;
+
+        /** match for each directive in the header */
+        type DirectiveMatch = { // eslint-disable-line no-unused-vars
+            'max-age': Array<string>,
+            'includesubdomains': Array<string>,
+            'preload': Array<string>
+        };
+
+        const loadRuleConfigs = () => {
+            minMaxAgeValue = (context.ruleOptions && context.ruleOptions.minMaxAgeValue) || 10886400; // 18 weeks
+        };
+
+        // STS header Syntax:
+        // Strict-Transport-Security: max-age=<expire-time>
+        // Strict-Transport-Security: max-age=<expire-time>; includeSubDomains
+        // Strict-Transport-Security: max-age=<expire-time>; preload
+        // This function accomplishes the following:
+        // "max-age=31536000; includesubdomains; preload" => {"max-age":31536000,"includesubdomains":true,"preload":true}
+        const parse = (headerValue: string) => {
+            const parsedHeader = {};
+            const directives = headerValue.toLowerCase().split(';');
+            const nameValuePairRegex = /^ *([!#$%&'*+.^_`|~0-9A-Za-z-]+) *= *("(?:[~0-9])*"|[!#$%&'*+.^_`|~0-9]+) *$/;
+            // Regex for name-value pairs. E.g.: max-age=31536000
+            // Modified usage of https://github.com/jshttp/content-type/blob/64bde0d996ccb4334341662c0c7d25f7b370c4d9/index.js#L23
+            const tokenRegex = /^ *[!#$%&'*+.^_`|~0-9A-Za-z-]+$/;
+            // Regex for single tokens. E.g.:  includesubdomains
+
+            directives.forEach((directive) => {
+                const match = tokenRegex.exec(directive) || nameValuePairRegex.exec(directive);
+
+                if (!match) {
+                    throw new Error(`'strict-transport-security' header has the wrong format: ${directive}`);
+                }
+
+                const [matchString, key, value] = match;
+                const name = key || matchString.trim();
+
+                if (parsedHeader[name]) {
+                    throw new Error(`'strict-transport-security' header contains more than one '${name}'`);
+                }
+
+                parsedHeader[name] = value || true;
+            });
+
+            return parsedHeader;
+        };
+
+        const isInvalidValue = (maxAge: string, limit: number): boolean => {
+            return maxAge && parseInt(maxAge) < limit;
+        };
+
+        const validate = async (fetchEnd: IFetchEnd) => {
+            const { element, resource, response }: { element: IAsyncHTMLElement, resource: string, response: IResponse } = fetchEnd;
+            const headerValue: string = normalizeString(response.headers && response.headers['strict-transport-security']);
+            let parsedHeader;
+
+            if (!isHTTPS(resource)) {
```

Comment:
@molant @alrra Yes.  The HTTP requests will automatically be converted to HTTPS. Quote from https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security:
>The first time your site is accessed using HTTPS and it returns the Strict-Transport-Security header, the browser records this information, so that future attempts to load the site using HTTP will automatically use HTTPS instead.

Shouldn't we just use the url after redirect? if it is `https`, check for the header, right? Does this mean jsdom doesn't do automatic redirect?
>Also I'm not sure the jsdom collector does it.